vowpalwabbit/cbify.cc

```diff
@@ -64,6 +73,42 @@ void finish(cbify& data)
   delete_it(data.mwt_explorer);
   delete_it(data.recorder);
   data.a_s.delete_v();
+  if (data.use_adf)
+  { for (size_t a = 0; a < data.adf_data.num_actions; ++a)
+    { VW::dealloc_example(CB::cb_label.delete_label, data.adf_data.ecs[a]);
```

Comment:
It's computationally inelegant to dealloc a large number of examples often.   Can you tweak the code to just reuse the examples instead?  That's what the rest of VW does.

Reply:
Doesn't the `finish` function here just get called on termination (through `set_finish`)? I am allocating the examples once on setup (see `init_adf_data`), reusing them on every example, and this call deallocs them.